autogpt/config/ai_config.py

```diff
@@ -79,11 +79,31 @@ def load(config_file: str = SAVE_FILE) -> "AIConfig":
 
         ai_name = config_params.get("ai_name", "")
         ai_role = config_params.get("ai_role", "")
-        ai_goals = config_params.get("ai_goals", [])
+        ai_goals = config_params.get("ai_role", "")
+        ai_goals = AIConfig.coerce_goals_to_str(config_params.get("ai_goals", []))
         api_budget = config_params.get("api_budget", 0.0)
         # type: Type[AIConfig]
         return AIConfig(ai_name, ai_role, ai_goals, api_budget)
 
+    @staticmethod
+    def coerce_goals_to_str(goals: list[Any]) -> list[str]:
+        """
+        Return goals as list of strings, in the format that the user intended,
+        fixing some cases where the goals are loaded from yaml as a list of dicts.
+
+        Parameters:
+            goals: The goals to sanitize (can be a string, dict, or other object)
+
+        Returns:
+            A list of strings, where each string is a goal.
+        """
+        to_str = (
```

Comment:
```suggestion
        goals = [str(goal).strip("{}").replace("'", "") if isinstance(goal, dict) else str(goal) for goal in goals]
```

Reply:
Like generally, there are some non-pythonic things here.  These are style things, and so not critical, but an explanation: 

- You don't typically want to assign lambda functions to variables.  The point of a lambda is that it's anonymous.  You should use the def syntax if you want to name a function. This is explicitly mentioned in pep8: https://peps.python.org/pep-0008/#programming-recommendations
- functional programming, while supported, isn't really first class in python.  You should strongly prefer list comprehensions over functional applications of map, reduce, filter, etc.  See https://www.artima.com/weblogs/viewpost.jsp?thread=98196 for some context.

Reply:
Sigh, there aren't many places where one gets to use both lambda and map in py 3 :)

I had implemented this PR as an inline list comprehension, but  it looked barely readable (I'd kept the k: v for k, v in... syntax), and for some reason I'd recalled (wrongly) that map() was faster. 

The strip("{}").replace() fixes readability. I'll probably just put it back inline then. The separate function seems redundant, unless you object.

Does the strip().replace() syntax have edge cases worth considering? I can think of, maybe, quotes in the string, but none where losing the " ' " will be catastrophic... 

"Goal: critique the first agent's output"

Good reads and insightful comments thanks for sharing. I don't think I've ever actually used reduce() 

I'll submit an update.